Orange/widgets/visualize/owmosaic.py

```diff
@@ -461,6 +462,9 @@ def set_subset_data(self, data):
         except:
             self.subset_data = None
             self.Warning.incompatible_subset(shown=data is not None)
+        finally:
+            indices = list(set(e.id for e in self.subset_data))
```

Comment:
`indices = {e.id for e in self.subset_data}`

That is, use set comprehension and, more importantly, `indices` should be a set.